### PR TITLE
cloudbuild: fix docs generation + adds testing for GCB triggers

### LINKED ISSUE #262: Use --cache-from in container builder
Container Builder does a full build each time. It would be interesting to conditionally use a `--cache-from` option to speed the build up. See https://cloud.google.com/container-builder/docs/speeding-up-builds

### LINKED ISSUE #668: Refactor skaffold docker image to use multi-stage build

### PR DESCRIPTION
**Pull request by balopat**
#648 broke the build - this fixes it
Fixes #262.
Fixes #668.

### CODE REVIEW

**Comment by balopat** [comment]
@r2d4 this should work, I'm running the last tests. 
We will have to introduce the `$SKAFFOLD_IMAGE` and `$SKAFFOLD_BUCKET` substitutions on the GCB triggers for this to work before merging to master!

**Comment by balopat** [comment]
FYI. added the substitution parameters in both triggers:

image

File: hack/gcb_testing/simulate_build_trigger.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+IMAGE=$1
+BUCKET=$2
+
+gcloud container builds submit . \
+    --config=deploy/cloudbuild.yaml \
+    --substitutions="_SKAFFOLD_IMAGE=$IMAGE,_SKAFFOLD_BUCKET=$BUCKET,COMMIT_SHA=testing_$(date +%s)"
```
Comment (r2d4) [inline hack/gcb_testing/simulate_build_trigger.sh]: I'd rather have this be driven off the makefile rather than cloudbuild substitutions.

**Comment by balopat** [comment]
Substitutions allow for testing the cloudbuild scripts on different projects from k8s-skaffold and the skaffold bucket. Otherwise it's impossible to test cloudbuilds without messing with production data.

File: hack/gcb_testing/simulate_build_trigger.sh (hunk 1 above)
Comment (balopat) [reply hack/gcb_testing/simulate_build_trigger.sh]: I am not sure what you mean - I create a target in the Makefile, but the substitutions would stay anyway - unless we want to test the pipelines only in prod.

File: Makefile [hunk 2]
```diff
@@ -81,9 +81,46 @@ integration: install $(BUILD_DIR)/$(PROJECT)
...
+	gsutil -m cp $(BUILD_DIR)/$(PROJECT)-* gs://$(RELEASE_BUCKET)/releases/$(VERSION)/
+	gsutil -m cp $(BUILD_DIR)/$(PROJECT)-* gs://$(RELEASE_BUCKET)/latest/
+
+.PHONY: release-build
+release-build: cross docs
+	docker build \
+    		-f deploy/skaffold/Dockerfile \
+    		--cache-from gcr.io/$(GCP_PROJECT)/skaffold-builder \
+    		-t gcr.io/$(GCP_PROJECT)/skaffold:$(COMMIT) .
+	gsutil -m cp $(BUILD_DIR)/$(PROJECT)-* gs://$(RELEASE_BUCKET)/builds/$(VERSION)/
```
Comment (balopat) [inline Makefile]: The folder should be $(COMMIT)

**Comment by balopat** [comment]
@r2d4 ready to go. 
Additional benefits: 
 * the build is 1 minute faster than usual even though it has the asciidocs 
 * it fails if there is a mismatch between the actual git tag and the Makefile tag :)

**Comment by balopat** [comment]
Added the `_RELEASE_BUCKET` substitution on both triggers as well - also tested the triggers themselves on my project - it would be great if we could create the triggers from code but it seems it's a UI only feature (probably due to the Github auth step).

**Comment by balopat** [comment]
with the latest updates the time for builds is between 5 min 13 sec and 3 min 54 sec on my project. It takes 2 min 30 secs to start to see the logs - that might be the time for spinning up the bigger machine, but still faster than 9 mins.

**Comment by r2d4** [comment]
Maybe squash a few of these commits and merge?

**Comment by Prasanna4743** [comment]
??=BD20$

**Pull request merged**